File: src/main/java/com/alkawallet/model/Wallet.java
package com.alkawallet.model;

public class Wallet {
    private double balance;

    public Wallet() {
        this.balance = 0.0;
    }

    public Wallet(double initialBalance) {
        this.balance = initialBalance;
    }

    public double getBalance() {
        return balance;
    }

    public void deposit(double amount) {
        if (amount > 0) {
            this.balance += amount;
        }
    }

    public boolean withdraw(double amount) {
        if (amount > 0 && this.balance >= amount) {
            this.balance -= amount;
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/alkawallet/model/Wallet.kt
package com.alkawallet.model

class Wallet(initialBalance: Double = 0.0) {
    var balance: Double = initialBalance
        private set

    fun deposit(amount: Double): Boolean {
        return if (amount > 0) {
            balance += amount
            true
        } else {
            false
        }
    }

    fun withdraw(amount: Double): Boolean {
        return if (amount > 0 && balance >= amount) {
            balance -= amount
            true
        } else {
            false
        }
    }
}
